draft-trammell-taps-post-sockets.md

```diff
@@ -701,23 +718,88 @@ func receiveMulticast() {
 }
 ~~~~~~~~
 
+## Association Bootstrapping
+
+Here, we show how Association state may be initialized. The goal is to
+create a long-term Association from which Carriers may be derived and,
+if possible, used immediately. We will describe two variations -- one
+where the Association is configured with trust model information
+necessary for the Remote to establish a connection, and another
+where the Association is created from a pre-shared key established
+out-of-band mechanism.
+
+### Association Preparation
+
+Perhaps the more common case for creating Associations is to initialize
+new connections to Remotes. To do so, Associations must possess the
+security information needed to create a fresh connection.
+Per {{I-D.ietf-taps-transport-security}}, this includes, at a minimum:
+
+- An identity and access or an interface to the associated private key.
+- Trust model constraints, such as pinned public keys and anchor certificates.
+
+The following example shows how an Association can be created with this
+information to be later used for creating a remote connection.
+
+~~~~~~~~
+// create an association with a given identity and set of trusted certificates
+func createAssociation(ident Identity, trustedCerts []Certificate) Association {
+    association := Association()
+    association = association.SetIdentity(ident)
+    association = association.SetTrustedCerts(trustedCerts)
+    return association
+}
+~~~~~~~~
+
+The following example shows how to bootstrap an Association with a
+pre-shared key (PSK) distributed or otherwise obtained through an out-of-band
+mechanism.
+
+~~~~~~~~
+// create an Association using a pre-shared key
+func createAssociationWithPSK(preSharedKey []byte) Association {
+    association := AssociationFromPSK(preSharedKey)
+    return association
+}
+~~~~~~~~
+
+### Creating Carriers with Associations
+
+In this example, we show how to create a Carrier from an existing, pre-configured
+Association.
+
+~~~~~~~~
+// resume a connection to a server, reusing an existing Association given a
+// remote, and send some 0-RTT data
+func sayHelloQuicklyWithAssociation(association Association) {
+    parameters := Parameters(local, association)
+    carrier := Initiate(parameters, remote)
```

Comment:
I think you would also give the carrier only the local and the remote in this case, and if an association already exists with this local and remote, the carrier would automatically use that. I guess you can 'manually' create an assotation first to make sure it already exists... or you can probably have a policy if you want to use a specific existing association or want to enforce to create a new assocition when you call Initiate. Does that make sense?